src/App.js

```diff
@@ -32,9 +35,23 @@ export default function ActivityApp() {
   //----- -----
   return (
     <Appcontainer>
-      <DurationFilter onFilterDurationReset={filterDurationReset} onFilterDurationValue={setDurationFilterValue} />
-      <TypeFilter onFilterTypeReset={filterTypeReset} onFilterTypeValue={setTypeFiltervalue} />
-      <ActivityCards activities={filteredAll} onSetActivities={setActivities} />
+      {isHidden === false && (
+        <FavoritesPage activities={activities} favorite={favorite} isHidden={isHidden} onSetIsHidden={setIsHidden} />
+      )}
+      {isHidden === true && (
+        <DurationFilter onFilterDurationReset={filterDurationReset} onFilterDurationValue={setDurationFilterValue} />
+      )}
+      {isHidden === true && <TypeFilter onFilterTypeReset={filterTypeReset} onFilterTypeValue={setTypeFiltervalue} />}
+      {isHidden === true && (
+        <ActivityCards
+          activities={filteredAll}
+          onSetActivities={setActivities}
+          favorite={favorite}
+          onSetFavorite={setFavorite}
```

Comment:
You should better don't lift your set function down.

Reply:
Some coaches say the states should always be in app.js and everything needed should be lifted down (state &setState). 

Here, I could place the button, that needs the state, in app.js and conditionally render it if FavoritePage is not hidden.

Reply:
Instead of passing the `setFunction` down, it is recommended to pass a callback function like `handleFavorite` and within that function call the `setFavorite` but in the `App.js` where your state is.